libafl/src/inputs/mod.rs

```diff
@@ -95,3 +95,109 @@ pub trait HasBytesVec {
     /// The internal bytes map (as mutable borrow)
     fn bytes_mut(&mut self) -> &mut Vec<u8>;
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::bolts::current_nanos;
+
+    #[cfg(feature = "nautilus")]
+    use grammartec::{context::Context, tree::Tree};
+
+    static TEST_INPUT: &str = "I'm a test case";
+
+    #[test]
+    /// test that `current_nanos` increment; basis of uniqueness assertion in `generate_name` methods
+    fn test_current_nanos_increment_on_each_call() {
+        // no appreciable time spent here, even tho iterations seem high
+        for _ in 1..1000000 {
+            let first = current_nanos();
+            let second = current_nanos();
+
+            assert!(first < second);
```

Comment:
CI says `current_nano() < current_nano()` is not always true.
maybe we can use client's id to make the filename unique?

Reply:
> CI says `current_nano() < current_nano()` is not always true.
maybe we can use client's id to make the filename unique?

yea, looks like it craps out on mac/no-std :disappointed: 

Where is the client's ID relative to where name generation occurs? Right now, the `.generate_name` method is called in `OnDiskCorpus.add`. That method has `self` and a `TestCase` available. I'm guessing you mean the sender ID from the event manager? If so, in what way would you want that info passed to the corpus/how can the corpus query that info?

Reply:
uh-oh, after I quickly looked into this, it's not that straight-forward I guess 🥺 
client_id's are something on the llmp/event-manager level and that are not accessible from OnDiskCorpus sadly, so my idea was too flimsy.

what I have come up with is to use `Uuid::new_v4();`, it's std only, but so is OnDiskCorpus 😉

Reply:
what do you think @domenukk @andreafioraldi

Reply:
Eventually we'll want a way to document mutations and feedback leading to a corpus, anyway, so we may need to do some reshuffliing

Reply:
We already have one to do that `LoggerScheduledMutator`, but it is optional to use it

Reply:
uuid should be fine for std builds, then, right?